src/app/mod.rs

```diff
@@ -9,7 +12,8 @@ pub struct App {
 
 impl App {
     pub fn new(api: String) -> anyhow::Result<Self> {
-        let client = GaloyClient::new(api, None)?;
+        let token = token::get_token()?;
```

Comment:
how can this pattern handle cases where we require the user to be logged in?
in such situations we need to show the user that the user should be logged in or pass in the token as an error instead of silently biting the token off and still proceeding with creating the client.